system/Session/Handlers/ArrayHandler.php

```diff
@@ -22,78 +22,70 @@ class ArrayHandler extends BaseHandler
     protected static $cache = [];
 
     /**
-     * Open
+     * Re-initialize existing session, or creates a new one.
      *
-     * Ensures we have an initialized database connection.
-     *
-     * @param string $savePath Path to session files' directory
-     * @param string $name     Session cookie name
-     *
-     * @throws Exception
+     * @param string $path The path where to store/retrieve the session
+     * @param string $name The session name
      */
-    public function open($savePath, $name): bool
+    public function open($path, $name): bool
     {
         return true;
     }
 
     /**
-     * Read
+     * Reads the session data from the session storage, and returns the results.
      *
-     * Reads session data and acquires a lock
+     * @param string $id The session ID
      *
-     * @param string $sessionID Session ID
-     *
-     * @return string Serialized session data
+     * @return false|string Returns an encoded string of the read data.
+     *                      If nothing was read, it must return false.
      */
-    public function read($sessionID): string
+    #[ReturnTypeWillChange]
+    public function read($id)
     {
         return '';
     }
 
     /**
-     * Write
-     *
-     * Writes (create / update) session data
+     * Writes the session data to the session storage.
      *
-     * @param string $sessionID   Session ID
-     * @param string $sessionData Serialized session data
+     * @param string $id   The session ID
+     * @param string $data The encoded session data
      */
-    public function write($sessionID, $sessionData): bool
+    public function write($id, $data): bool
     {
         return true;
     }
 
     /**
-     * Close
-     *
-     * Releases locks and closes file descriptor.
+     * Closes the current session.
      */
     public function close(): bool
     {
         return true;
     }
 
     /**
-     * Destroy
-     *
-     * Destroys the current session.
+     * Destroys a session
      *
-     * @param string $sessionID
+     * @param string $id The session ID being destroyed
      */
-    public function destroy($sessionID): bool
+    public function destroy($id): bool
     {
         return true;
     }
 
     /**
-     * Garbage Collector
+     * Cleans up expired sessions.
      *
-     * Deletes expired sessions
+     * @param int $max_lifetime Sessions that have not updated
+     *                          for the last max_lifetime seconds will be removed.
      *
-     * @param int $maxlifetime Maximum lifetime of sessions
+     * @return false|int Returns the number of deleted sessions on success, or false on failure.
      */
-    public function gc($maxlifetime): bool
+    #[ReturnTypeWillChange]
+    public function gc($max_lifetime)
```

Comment:
Any reason why not to use camelCase?

Reply:
I based it here https://www.php.net/manual/en/sessionhandlerinterface.gc.php

I'm not sure though if this will have a bearing on 8.1.

Reply:
> And finally, named arguments will deal in a pragmatic way with name changes during inheritance.

From https://stitcher.io/blog/php-8-named-arguments. There is a good example there but basically: you can change the names in child classes, but names arguments will always have to match the actual class's argument signature. 

I think matching this to its parent signature is a good idea.

Reply:
Ok, thank you both. 
Sorry, apparently I'm not up to date with the all-new features of the language 😓

Reply:
All good! It can be a side-effect of maintaining all supported versions: you're always working with the oldest feature set. I had a moment looking at the PHP version chart (https://www.php.net/supported-versions.php) where I was like "we should be on active support, not security support!" But in reality I think dropping 7.3 this year will already be a big enough change. Next year when we are `>= 8.0` the real fun will begin 🤗